src/commonMain/kotlin/org/kosat/Kosat.kt

```diff
@@ -2,12 +2,23 @@ package org.kosat
 
 import kotlin.math.abs
 
-class Kosat(clauses: MutableList<MutableList<Lit>>, vars: Int = 0): Solver {
+fun solveWithAssumptions(cnf: CnfRequest): List<Int>? {
+    val clauses = (cnf.clauses.map { it.lit }).toMutableList()
+    val solver = KoSat(clauses, cnf.vars)
+    return if (solver.solve()) solver.getModel() else null
+}
+
+fun solveCnf(cnf: CnfRequest): List<Int>? {
+    val clauses = (cnf.clauses.map { it.lit }).toMutableList()
+    return NonIncremental(clauses, cnf.vars).solve()
```

Comment:
Your interfaces are very inconsistent. Why with `Kosat` you have to check `if solver.solve()` and then query a model with `getModel`, but for `NonIncremental` you just delegate to `.solve()`?
(I know *why* – because you interfaces are so. This is a rhetorical question, as most of them above :) )